tasks/build/verify_translations.js

```diff
@@ -0,0 +1,57 @@
+import Promise from 'bluebird';
+
+import fromRoot from '../../src/utils/from_root';
+import KbnServer from '../../src/server/kbn_server';
+import * as i18nVerify from '../../src/core_plugins/i18n/server/tool/i18n_verify_keys';
+import * as i18n from '../../src/core_plugins/i18n/server/i18n/i18n';
+
+module.exports = function (grunt) {
+  grunt.registerTask('_build:verifyTranslations', async function () {
+    const parsePaths = [fromRoot('/src/ui/views/*.jade')];
+    const keyPattern = 'i18n\\(\'(.*)\'\\)';
+    const keyPatternRegEx = new RegExp(keyPattern, 'g');
+
+    const serverConfig = {
+      env: 'production',
+      logging: {
+        silent: true,
+        quiet: true,
+        verbose: false
+      },
+      optimize: {
+        useBundleCache: false,
+        enabled: false
+      },
+      server: {
+        autoListen: false
+      },
+      plugins: {
+        initialize: true,
+        scanDirs: [fromRoot('src/core_plugins')]
+      },
+      uiSettings: {
+        enabled: false
+      }
+    };
+    const kbnServer = new KbnServer(serverConfig);
+    await kbnServer.ready();
+
+    const locales = i18n.getRegisteredTranslationLocales();
+    Promise.all(locales.map(function (locale) {
+      return i18n.getTranslationsForLocale(locale).then(function (translations) {
+        i18nVerify.verifyTranslationKeys(parsePaths, translations, keyPatternRegEx).then(function (keysNotTranslated) {
+          if (keysNotTranslated && Object.keys(keysNotTranslated).length > 0) {
+            console.error('Verification of the following translations keys are not translated: ', keysNotTranslated);
+          } else {
+            console.log('Verification of translations keys are a success.');
+          }
+        });
+      });
+    })).then(async function() {
+      await kbnServer.close();
+    }).catch(async function(e) {
+      await kbnServer.close();
+      throw e;
+    });
```

Comment:
Grunt doesn't understand promises. The `async` keyword is just syntactic sugar around promises so making the grunt task function `async` doesn't mean grunt will handle it properly as an async task. In order for grunt to understand that a task is async and needs to be treated specially, you have to call `const done = this.async();` and then call `done()` when you're done. Otherwise grunt will move on to other tasks before your async task is done, running things out of order. You can observe this when running `grunt build` in this PR. The verification log message gets spit out in the middle of a random task whenever the async task happens to finish.